### PR TITLE
Feat/user status modal

### PR DESCRIPTION
**Pull request by heyrandhir**
Closes all the points in https://github.com/Real-Dev-Squad/website-my/issues/340 except the 3rd one realted to future status.

- [x] Currently, Users can select any date while marking themselves as `Idle or Active`. We need to remove the `from date` field for `Idle or Active`. The `from date` field value will be the date on which they mark themselves as `Idle or Active`.  
- [x] Submit Button to enable only when all the mandatory fields are selected 
- [ ] If the users mark themselves `OOO` for a Future Date. It overrides the current status and shows `You are OOO`. It should show `OOO` only once the date arrives. Handle this case and also figure out what happens once the user is back in the office. 
- [x] If the Users are `OOO` for less than 3 days `Reason` shouldn't be mandatory. 
- [x] Disable past date selection for `OOO`

The remaining point 3 will be fixed in the future PR

[screen-capture.webm](https://user-images.githubusercontent.com/97341921/208008851-eb47f3c0-96dd-4b9c-865a-776f043f943e.webm)

### CODE REVIEW

File: app/components/user-status-modal.js [hunk 1]
```diff
@@ -88,6 +96,46 @@ export default class FormStatusModal extends Component {
...
+  }
+
+  @action
+  checkIfFromToDatesAreClose() {
+    if (this.fromDate && this.untilDate) {
+      let from = new Date(this.fromDate.replaceAll('-', ',')).getTime();
+      let until = new Date(this.untilDate.replaceAll('-', ',')).getTime();
+      const timeGap = until - from;
+      // 172800000 is the no of milli seconds in 3 days. as reason field is not mandatory if its less than 3 days
+      return timeGap <= 172800000;
```
Comment (rohan09-raj) [inline app/components/user-status-modal.js:136]: Can we have it in constants please ?

File: app/components/user-status-modal.hbs [hunk 2]
```diff
@@ -2,31 +2,30 @@
...
-      {{#if (not-eq @newStatus 'ACTIVE')}}
-      {{#if (eq @newStatus 'OOO')}}
+      <Input name='untilDate' type='date' min={{if this.fromDate this.fromDate this.disableDatesPrior}}
+        value={{this.untilDate}} {{on "change" this.updateValue}} data-test-date-picker-until />
       <label class="modal__inputs--description" data-test-label-reason>Reason</label>
-      {{/if}}
 
-      {{#if (eq @newStatus 'IDLE')}}
-      <label class="modal__inputs--description" data-test-label-missing-skills>What skills are you willing to learn</label>
+      {{else if (eq @newStatus 'IDLE')}}
```
Comment (rohan09-raj) [inline app/components/user-status-modal.hbs:15]: You can use the constants here for `IDLE` by defining the constant in the component file.

File: app/components/user-status-modal.hbs [hunk 3]
```diff
@@ -2,31 +2,30 @@
...
+
+      <label class="modal__inputs--description" data-test-label-missing-skills>What skills are you willing to
+        learn</label>
+
```
Comment (rohan09-raj) [inline app/components/user-status-modal.hbs:19]: Can we remove this extra line spaces?

File: app/components/user-status.js [hunk 4]
```diff
@@ -53,4 +54,20 @@ export default class UserStatusComponent extends Component {
       ],
     },
   ];
+
+  @action async changeStatus(status) {
+    if (status === USER_STATES.ACTIVE) {
+      const updatedAt = Date.now();
+      const activeStateObj = {
```
Comment (rohan09-raj) [inline app/components/user-status.js:61]: Can we name as `activeState` only as it's obvious that it's a object by reading the code

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

**Review by MehulKChaudhari — APPROVED** [review]

File: app/components/user-status-modal.js [hunk 5]
```diff
@@ -88,6 +98,45 @@ export default class FormStatusModal extends Component {
...
+
+  @action
+  handleInput(event) {
+    const { value } = event.target;
+    this.reason = value;
+    this.checkSubmitBttnState();
+  }
+
+  @action
+  checkSubmitBttnState() {
```
Comment (rohan09-raj) [inline app/components/user-status-modal.js:112]: Type : Bttn => Btn

File: app/components/user-status-modal.js [hunk 6]
```diff
@@ -30,24 +33,22 @@ export default class FormStatusModal extends Component {
     } else if (name === 'reason') {
       this.reason = value;
     }
+    this.checkSubmitBttnState();
```
Comment (rohan09-raj) [inline app/components/user-status-modal.js:36]: Type : Bttn => Btn

File: app/components/user-status-modal.js [hunk 7]
```diff
@@ -30,24 +33,22 @@ export default class FormStatusModal extends Component {
     } else if (name === 'reason') {
```
Comment (rohan09-raj) [inline app/components/user-status-modal.js:33]: I think we can use constants here.

**Review by rohan09-raj — APPROVED** [review]
Great work 🔥

**Comment by heyrandhir** [comment]
the 3rd point `If the users mark themselves OOO for a Future Date. It overrides the current status and shows You are OOO. It should show OOO only once the date arrives. Handle this case and also figure out what happens once the user is back in the office.` changes are active in the backend.

**Pull request merged**